### ISSUE TITLE
Files Sharing javascript cleanup

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#874]
As a leftover of times long gone there is still a lot of sharing code all over te place. This issue focues on the javascript part.

* Sidebar code is in the files_sharing app
* ShareModel and views are in core
* ShareModel depends on the OCS Sharing API which lives in files_sharing

We should define a proper separation between what is in core and what is not. Kind of like we do on the PHP side as well. The sharemanager is in core. But the share providers (should) live in the different apps. We could do the same with the javascript:

* Have a kind of share manager where apps can register
  * This is just a collection of collections
* Items models live in their own apps
* Views don't live in core since each app might want to display different stuff.
* We also need to split sharee loading then??

In short this makes the js part somewhat modular. If an other app wants to use sharing functionality it can use the 'manager' in OC.Share.ShareManager (or whatever). And trough that it will automagically get access to all the fancy stuff it can do.

CC: @oparoz as this is kind of your JS API you have been asking for

CC: @schiessle @nickvergessen @MorrisJobke @icewind1991 any input idea's?

**Comment by MorrisJobke** [comment]
As alternative: Do we want to build a unique way of only accessing this via OCS and don't bind it too much together (front and backend). Or is this already the case?

**Comment by rullzer** [comment]
Well you need to somewhat bind it since the API endpoint is not very intuitive. Lots of deps on how you add a type of share etc. E.g. link shares are handled different from all the others. Federated shares are still different from the other stuff.

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!
